### DISCUSSION TITLE
Are we going to use Async Traits in actix-web (version 5!)

### DISCUSSION THREAD

**Discussion by omid** [actix/actix-web#3242] [General] [answered]
Full support of Async Trait is around the corner.

Are there any plan or ideas to use it in actix-web?
For handlers maybe?
Or ServiceConfig?

**Comment by robjtede** [comment, marked as answer]
Yah probably. Started some drafts of using it here <https://github.com/actix/actix-web/pull/3192>.

**Reply by omid** [reply]
Thanks.
As far as I can see, it's about migrating `FromRequest`, so far.
Are there any other plans to change the API?

**Reply by robjtede** [reply]
`FromRequest` is one of the obvious targets.
Do you have any other ideas?
E.g., you mentioned `ServiceConfig`, what's the idea there?
